Allow users to directly set feature output column names and save during serialization

thehomebrewnerd opened the pull request:
### Allow users to directly set feature output column names and save during serialization

Adds a `set_feature_names` method to `FeatureBase` to directly allow for setting of feature column names to override the default name. Also updates serialization to save and restore these names.

rwedge commented on featuretools/tests/computational_backend/test_calculate_feature_matrix.py:
```diff
@@ -2266,3 +2275,25 @@ def multi_cum_sum(x):
+    )
+
+    direct_name = ["device_direct"]
+    direct_feat.set_feature_names(direct_name)
```
for single output features, do we want `set_feature_names` and `rename` to both work?

thehomebrewnerd replied:
Not sure - are you suggesting that we just allow `set_feature_names` to work for multi-output features and renaming feature columns for single output should be done through `rename` as you can be done currently?

thehomebrewnerd replied:
As I think about it I think I like the idea of having `set_feature_names` operate only on multi-output features - seems more consistent with the method name, and would reduce the scope of the change. We could add a check in the method to raise an error if `number_output_features == 1`.

@rwedge What do you think about that approach?

thehomebrewnerd replied:
@rwedge Sounds good. I'll make this change and add the additional test case you recommended.

rwedge reviewed (commented):
Can we test that a feature stacked on a renamed column from a feature with multiple outputs uses the renamed name?

thehomebrewnerd commented:
> Can we test that a feature stacked on a renamed column from a feature with multiple outputs uses the renamed name?

@rwedge I added a test that checks the name of a feature created by stacking on a `FeatureOutputSlice` of a multi-output feature. Also updated tests to include a multi-output GroupbyTransformFeature.

rwedge commented on featuretools/feature_base/feature_base.py:
```diff
@@ -107,19 +107,26 @@ def get_feature_names(self):
         Number of names provided much match the number of output columns defined for
-        the feature.
+        the feature. Only works for features that have more than one output column. Use
+        ``Feature.rename`` for changing the column name for single output features.
```
```suggestion
        ``Feature.rename`` to change the column name for single output features.
```
